Add Search and Replace dialog window

ukito-pl opened the pull request:
Adds QCodeEditorTextFinder class that creates a dialog window and allows user to search and replace text in editor.

NicolasKogler commented:
Hello,

please re-check the files you have changed - it seems you have converted all the project's files from Unix line endings (LF)  to Windows line endings (CRLF). You may have converted spaces to tabs too.

Furthermore, please adapt your actual changes to the code style of this project, mainly:
* Group local variables, separate code with empty lines
* Pointer or reference specifiers belong to the type, not to the variable
* Insert a space character between braces and expressions
* Members are prefixed with _m\__,  followed by the lowercase camelCase name (e.g. m_parentWidget)

```cpp
// Wrong (braces & reference)
void QCodeEditorTextFinder::onTextToReplaceChanged(const QString &text){
    if (!text.isEmpty()){

// Correct
void QCodeEditorTextFinder::onTextToReplaceChanged(const QString& text) {
    if (!text.isEmpty()) {

// Wrong (spaghetti code)
findDialog = new QDialog(this);
findDialog->setWindowTitle("Find and replace");
findDialog->resize(550,100);
dialogLayout = new QGridLayout;
replaceButton = new QPushButton("Replace");
replaceAllButton = new QPushButton("Replace all");

// Correct
findDialog = new QDialog(this);
findDialog->setWindowTitle("Find and replace");
findDialog->resize(550,100);

dialogLayout = new QGridLayout;
replaceButton = new QPushButton("Replace");
replaceAllButton = new QPushButton("Replace all");
```

Thank you in advance!

KR
Nicolas Kogler

ukito-pl commented:
Thank you very much for such precise and helpful response. 

I think I fixed the CRLF/LF confusion and also changed code style of my changes according to your points. Hope it looks better now. :)

With regards,

Marek Buła

NicolasKogler commented:
Thank you for enforcing the coding guidelines I provided, Marek. I now merged your request! :)

KR
Nicolas Kogler
